feat: clearer backup and restore info

rolznz opened the pull request:
Fixes https://github.com/getAlby/hub/issues/645

Adds a link to our guide, makes copy on the backup and seed imports more clear and warns against importing from seed if you have open channels

![image](https://github.com/user-attachments/assets/1121bc6d-f3e8-4b3c-a204-23f6736d22a1)

![image](https://github.com/user-attachments/assets/b7a44183-3858-4adc-b4ac-9326fce1085d)

im-adithya commented on frontend/src/screens/setup/ImportMnemonic.tsx:
```diff
@@ -90,6 +95,16 @@ export function ImportMnemonic() {
+                channels. Please contact Alby support before proceeding if you
+                had channels on a different device.
+              </span>
+            </div>
```
I really like those checkboxes above which re-confirm the warnings. Should we add a similar check box here to tell the lightning channel balance is not restored?

(My understanding is that if the user ignores the warnings and continues then they might not be able to recover the lightning balance(?) If yes then it's good to add one here too)

rolznz replied:
Good idea, thanks

reneaaron commented on frontend/src/screens/BackupMnemonic.tsx:
```diff
@@ -149,34 +157,64 @@ export function BackupMnemonic() {
+                <b>recovery phrase</b>&nbsp;
+                {info?.albyAccountConnected && (
+                  <>
+                    or your <b>unlock password</b>
```
!(recovery phrase || unlock) vs !(recovery phrase && unlock)

rolznz replied:
I'm not sure this is right, can you check the screenshot? it says if you forget either of them, you will lose funds.

Currently: If you forget the unlock password, you lose funds from channels. If you forget the seed, you lose everything.

im-adithya replied:
> I'm not sure this is right, can you check the screenshot?

+1

> and do not have your recovery phrase or your unlock password

=> do not have recovery phrase or do not have unlock password
=> `!recovery_phrase || !unlock_password`
=> `!(recovery phrase && unlock_password)`

rolznz replied:
@im-adithya I don't understand the second part, do you think we need to do anything here or just resolve this thread?

im-adithya replied:
No no it's good, we can resolve this 👍

bumi commented on frontend/src/screens/BackupMnemonic.tsx:
```diff
@@ -149,34 +157,64 @@ export function BackupMnemonic() {
+              />
+              <Label htmlFor="backup2" className="ml-2">
+                I understand the recovery phrase alone will not restore funds
+                from my lightning channels.{" "}
```
can we say this active with the necessary details? 

I understand the unlock password AND the recovery phrase is needed to recover funds
from my lightning channels.{" "}

something like this? 

otherwise you read it and at that point don't get the necessary info.

jpl-btc replied:
People tend to remember the words they read, and positive affirmations resonate better than negative ones, aligning perfectly with Bumi's suggestion.

rolznz commented:
Addressed feedback:
- More clear text in second checkbox on backup page, now also only shows when first checkbox is checked
- Added checkbox on import page

jpl-btc commented:
Alternative copy for the 3 warning sentences (feel free to use/remix/not use):

- Your recovery phrase is a set of 12 words used to restore your savings balance from a backup.  

- Keep it safe and private to ensure your funds remain secure.

- Please note, the recovery phrase cannot restore funds from lightning channels. If you had active channels on a different device, contact Alby support before proceeding.  (note: the red color is not eye friendly and I have encountered colorblind users that had to select the text with the mouse to be able to read it, this happens also when setting up the unlock password)

rolznz commented:
@jpl-btc thanks for the feedback. Updated the copy and removed the red styling on the pages. The setup password second confirmation checkbox has bold text which differentiates it from the first confirmation box anyway, I think that's enough (and the destructive style was a bit buggy anyway. CC @reneaaron )
